contracts/core/Controller.sol

```diff
@@ -324,14 +323,24 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
         emit Donated(msg.sender, _asset, _amount);
     }
 
+    /**
+     * @notice send asset amount to margin pool v2
+     * @dev use donate() instead of direct transfer() to store the balance in assetBalance
+     * @param _asset asset address
+     * @param _amount amount to donate to pool
+     */
+    function donateV2(address _asset, uint256 _amount) external {
+        poolV2.transferToPool(_asset, msg.sender, _amount);
+
+        emit Donated(msg.sender, _asset, _amount);
+    }
+
     /**
      * @notice allows the partialPauser to toggle the systemPartiallyPaused variable and partially pause or partially unpause the system
      * @dev can only be called by the partialPauser
      * @param _partiallyPaused new boolean value to set systemPartiallyPaused to
      */
     function setSystemPartiallyPaused(bool _partiallyPaused) external onlyPartialPauser {
-        require(systemPartiallyPaused != _partiallyPaused, "C9");
```

Comment:
IMO no harm for keeping those and it make it feel safer, also the owner is the only one capable of calling this function so it is fine to pay for extra gas

Reply:
I would have left it, but otherwise I run into contract size issues and this seemed like the least consequential to remove.